screen/wallets/OnboardingScreen.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useCallback } from "react";
+import { View, Text, TouchableOpacity, StyleSheet, Image } from 'react-native';
+import { useExtendedNavigation } from '../../hooks/useExtendedNavigation';
+import { DetailViewStackParamList } from "../../navigation/DetailViewStackParamList";
+import { NativeStackNavigationProp } from '@react-navigation/native-stack';
+import { useTheme, RouteProp } from "@react-navigation/native";
+import { HDSegwitBech32Wallet } from '../../class/index';
+import loc from '../../loc';
+import { useStorage } from '../../hooks/context/useStorage';
+import triggerHapticFeedback, { HapticFeedbackTypes } from '../../blue_modules/hapticFeedback';
+
+type NavigationProps = NativeStackNavigationProp<DetailViewStackParamList, 'WalletsList'>;
+type RouteProps = RouteProp<DetailViewStackParamList, 'WalletsList'>;
+
+const OnboardingScreen: React.FC = () => {
+
+  const navigation = useExtendedNavigation<NavigationProps>();
+  const { colors } = useTheme();
+  const { addWallet, saveToDisk } = useStorage();
+
+  const createWallet = async () => {
+    try {
+      // Create a new HDSegwitBech32Wallet (native segwit) directly
+      const w = new HDSegwitBech32Wallet();
```

Comment:
now that the address generation pr is in, we should be creating a HDSilentPaymentsWallet
```suggestion
      const w = new HDSilentPaymentsWallet();
```